### DISCUSSION TITLE
Query param without value

### DISCUSSION THREAD

**Discussion by crystalfp** [pactumjs/pactum#231] [Q&A]
If I want to pass a boolean query parameter to a GET request like:
`http://localhost/api/users?refresh` that is, without passing a value to the query parameter, I should do this:
```
await spec()
  .get(endpointUsers)
  .withQueryParams("refresh", "")
  .expectStatus(200)
  .toss();
```
That call generates the following GET: `http://localhost/api/users?refresh=`

Do you think could be a good idea to modify ` .withQueryParams()` to accept the key only and generate the original GET? I know, it is a minor feature, but cost almost nothing to implement.

Thanks for considering!
mario

**Comment by ASaiAnudeep** [comment]
I think we can mention the partial query param in the `url`.

```js
await spec()
  .get(endpointUsers + '?refresh')
  .expectStatus(200)
  .toss();
```

**Reply by crystalfp** [reply]
The suggestion does not work if you have also other  `.withQueryParams()` params.
For example, this code:
```
	await spec()
		.get("http://localhost/api/kb?rank")
		.withQueryParams("query", "cosa")
		.withQueryParams("lang", "it")
```
generates the following invalid URL:
``` 
"http://localhost/api/kb?rank?query=cosa&lang=it",
```

**Reply by crystalfp** [reply]
I know, I could build the query string myself:
```
await spec()
	.get("http://localhost/api/kb?rank&query=cosa&lang=it")
```
But it means to ignore a useful spec method that has a simple fix.

The simplest fix is in `node_modules\pactum\src\helpers\helper.js` adding after line 33:
```
if(query[prop] === "")
        plainQuery = plainQuery + prop;
else
```
A more clear solution is to change also `Spec.js` and `Spec.d.ts` to accept a missing value internally transforming it into ""
What do you think?

**Reply by ASaiAnudeep** [reply]
This feature is released in [v3.2.4](https://github.com/pactumjs/pactum/releases/tag/v3.2.4)

**Reply by crystalfp** [reply]
Fantastic! Thanks for the very fast response.
The only very minor thing is that this parameter is not listed in the `.inspect()` output under `queryParams`:
```
  console.warn
    {
      "url": "http://localhost/api/kb?query=rosa&lang=it&rank",
      "method": "GET",
      "queryParams": {
        "query": "rosa",
        "lang": "it"
      },
      "path": "/api/kb",
      "timeout": 3000
    }
```
Anyway, thanks for the functionality! And don't forget to update the documentation at `https://pactumjs.github.io/api/requests/withQueryParams.html`
mario
